api/src/main/java/org/openmrs/module/registrationcore/api/mpi/common/MpiPatientFetcher.java

```diff
@@ -9,8 +10,16 @@
 
     /**
      * Perform query to mpi server to get patient by identifier & convert it to local patient.
```

Comment:
How are you going to convert the query results to a local patient for patients that don't exist locally?

I think we can have a special new transient class to represent the search results rather using Patient, you could add a convenience method that can transform it to a Patient object on demand.

Reply:
The patient does not need to exist locally in the DB for the query to be converted into a patient object. An example of where this is done is in the [PixPdqMessageUtil](https://github.com/shaoyuancc/openmrs-module-registrationcore/blob/f85bfa62042e54df0fff7c216506cd3b8d9d5afd/api/src/main/java/org/openmrs/module/registrationcore/api/mpi/pixpdq/PixPdqMessageUtil.java#L206)

Reply:
But as a consumer of this method, when you get me back the matched patient object I'd expect it to be a Patient record with a database id, what you get back from the mpi doesn't have a database id.

Reply:
If you convert the patient to a Patient object, it implies it wouldn't work for one that doesn't exist locally but does in the mpi, unless the querying code automatically imports them in case they don't exist locally upon a search which personally I'd think should not happen, fetching should only be done when explicility requested.

Reply:
Oh you are right, I just re-read the Javadoc description which says that this method converts it to a local patient (which it does not) furthermore this conflicts which the return annotation which says "converted patient not yet saved to local DB." 
So I think the answer to this is to change the comment to make it clear that this method does not do the conversion to local patient, it just puts it in the form of a Patient Object.

Reply:
What I'm saying is that if this method returns a core Patient object, an API consumer would think it's a saved instance with a patient Id and other fields specific that exist ONLY in OpenMRS, otherwise you might as well return  an instance of a custom Patient class defined in the module to avoid confusion.

Reply:
Okay sorry, I think I understand you now... should we return this [mpiPatient class](https://github.com/shaoyuancc/openmrs-module-registrationcore/blob/f85bfa62042e54df0fff7c216506cd3b8d9d5afd/api/src/main/java/org/openmrs/module/registrationcore/api/mpi/common/MpiPatient.java#L9) instead?

Reply:
Better!

Reply:
Wyclif, theres an inconsistency with that idea though. If Patient Objects are only for patients that have been saved locally, then during the construction of the patient from the registrationData in the [registerPatient method](https://github.com/shaoyuancc/openmrs-module-registrationcore/blob/f85bfa62042e54df0fff7c216506cd3b8d9d5afd/api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java#L182) the object used should not be a patient? it only truly becomes a patient after the savePatient method is called [here](https://github.com/shaoyuancc/openmrs-module-registrationcore/blob/f85bfa62042e54df0fff7c216506cd3b8d9d5afd/api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java#L238)

am I right?